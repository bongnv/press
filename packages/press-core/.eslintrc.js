const baseConfig = require("../../.eslintrc");

module.exports = {
  ...baseConfig,
  extends: [...baseConfig.extends, "plugin:@typescript-eslint/recommended"],
  parser: "@typescript-eslint/parser",
  plugins: [...baseConfig.plugins, "@typescript-eslint"],
  ignorePatterns: ["/lib/"],
};
