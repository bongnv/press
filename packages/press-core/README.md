# `@bongnv/press-core`

`press-core` contains core logic of `press`.
