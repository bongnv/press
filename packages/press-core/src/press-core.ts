import { Config, Execution } from "./execution";
import { applyPlugins } from "./apply-plugins";

function registerHooksExecution({ commands }: Execution): void {
  commands.build.tapPromise("HooksExecution", async (execution: Execution) => {
    const { hooks } = execution;
    await hooks.prepare.promise(execution);
    await hooks.configWebpack.promise(execution);
    await hooks.bundle.promise(execution);
    await hooks.generate.promise(execution);
  });

  commands.dev.tapPromise("HooksExecution", async (execution: Execution) => {
    const { hooks } = execution;
    await hooks.prepare.promise(execution);
    await hooks.configWebpack.promise(execution);
    await hooks.bundle.promise(execution);
  });
}

function prepareExecution(config: Config, execution: Execution): void {
  applyPlugins(config.plugins || [], execution);
  registerHooksExecution(execution);
}

export async function build(config: Config): Promise<void> {
  console.log("running build", config);
  const execution = new Execution(config);
  prepareExecution(config, execution);
  await execution.commands.build.promise(execution);
}

export async function dev(config: Config): Promise<void> {
  console.log("running dev", config);
  const execution = new Execution(config);
  prepareExecution(config, execution);
  await execution.commands.dev.promise(execution);
}

export type { Config } from "./execution";
