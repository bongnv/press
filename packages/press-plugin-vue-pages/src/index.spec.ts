import path from "path";
import { Execution } from "@bongnv/press-core";

import vuePagesPlugin from "./index";

test("Enhancers should apply configWebpack properly", async () => {
  jest
    .spyOn(path, "resolve")
    .mockImplementationOnce((_, localPath) => path.join("/", localPath));

  const execution = new Execution({
    baseDir: "/",
  });

  vuePagesPlugin(execution);
  await execution.hooks.configWebpack.promise(execution);
  expect(execution.clientWebpackConfig.toConfig()).toMatchSnapshot();
  expect(execution.serverWebpackConfig.toConfig()).toMatchSnapshot();
});
