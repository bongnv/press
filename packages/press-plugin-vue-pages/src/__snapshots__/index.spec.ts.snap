// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Enhancers should apply configWebpack properly 1`] = `
Object {
  "module": Object {
    "rules": Array [
      Object {
        "test": "/vue-app/routes-loader.js",
        "use": Array [
          Object {
            "loader": "val-loader",
            "options": Object {
              "baseDir": "/",
            },
          },
        ],
      },
    ],
  },
}
`;

exports[`Enhancers should apply configWebpack properly 2`] = `
Object {
  "module": Object {
    "rules": Array [
      Object {
        "test": "/vue-app/routes-loader.js",
        "use": Array [
          Object {
            "loader": "val-loader",
            "options": Object {
              "baseDir": "/",
            },
          },
        ],
      },
    ],
  },
}
`;
